test/integration/testpaths/testpath_attach_disk_zwps.py

```diff
@@ -0,0 +1,185 @@
+            cls.testdata["ostype"])
+        cls.skiptest = False
+
+        try:
+            cls.pools = StoragePool.list(cls.apiclient, zoneid=cls.zone.id)
+        except Exception as e:
+            cls.skiptest = True
+            return
+        try:
+
+            # Create an account
+            cls.account = Account.create(
+                cls.apiclient,
+                cls.testdata["account"],
+                domainid=cls.domain.id
+            )
+            cls._cleanup.append(cls.account)
+
+            # Create user api client of the account
+            cls.userapiclient = testClient.getUserApiClient(
+                UserName=cls.account.name,
+                DomainName=cls.account.domain
+            )
+            # Create Service offering
+            cls.service_offering_zone1 = ServiceOffering.create(
+                cls.apiclient,
+                cls.testdata["service_offering"],
+                tags=ZONETAG1
+            )
+            cls._cleanup.append(cls.service_offering_zone1)
+
+            # Create Disk offering
+            cls.disk_offering = DiskOffering.create(
+                cls.apiclient,
+                cls.testdata["disk_offering"]
+            )
+
+            cls._cleanup.append(cls.disk_offering)
+
+        except Exception as e:
+            cls.tearDownClass()
+            raise e
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+
+    def tearDown(self):
+        try:
+            if hasattr(self, "data_volume_created"):
+                data_volumes_list = Volume.list(
+                    self.userapiclient,
+                    id=self.data_volume_created.id,
+                    virtualmachineid=self.vm.id
+                )
+                if data_volumes_list:
+                    self.vm.detach_volume(
+                        self.userapiclient,
+                        data_volumes_list[0]
+                    )
+
+                status = validateList(data_volumes_list)
+                self.assertEqual(
+                    status[0],
+                    PASS,
+                    "DATA Volume List Validation Failed")
+
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags=["basic", "advanced"], required_hardware="true")
+    def test_01_attach_datadisk_to_vm_on_zwps(self):
+        """ Attach Data Disk To VM on ZWPS
+            1.   Get zwps storage pool.
+            2.   Attach data disk to vm which is on zwps.
+            3.   Verify disk is attached.
+        """
+

```

Comment:
Added missed code to check Zone wide primary storage and adding tags to storage pools with scope="Zone"